phoenix-core/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java

```diff
@@ -165,7 +183,7 @@ public RegionScanner preScannerOpen(final ObserverContext<RegionCoprocessorEnvir
             scan.setTimeRange(timeRange.getMin(), Bytes.toLong(txnScn));
         }
         if (isRegionObserverFor(scan)) {
-            if (! skipRegionBoundaryCheck(scan)) {
+            if (! skipRegionBoundaryCheck(scan) || ScanUtil.isLocalIndex(scan)) {
```

Comment:
Add comment, please. Something like this:

    // For local indexes, we need to throw if out of region as we'll get duplicates otherwise while in other cases,
    // it may just mean out client-side data on region boundaries is out of date and can safely be ignored.

Note that we'll need to handle this from the MR integration which might be painful.